src/services/twitter.service.ts

```diff
@@ -0,0 +1,118 @@
+import Environment from "../environment";
+import axios, { AxiosRequestConfig } from 'axios';
+
+export type Tweet = {
+  id: string;
+  text: string;
+  author_id: string;
+  created_at: string;
+  attachments?: {
+    media_keys: string[];
+  };
+  public_metrics: TwitterPublicMetrics;
+}
+
+export type TwitterPublicMetrics = {
+  retweet_count: number;
+  reply_count: number;
+  like_count: number;
+  quote_count: number;
+}
+
+export type TwitterUser = {
+  username: string,
+  profile_image_url: string,
+  id: string,
+  name: string,
+}
+
+type TwitterMedia = {
+  type?: 'photo' | 'animated_gif';
+  media_key?: string;
+  preview_image_url?: string;
+  url?: string;
+}
+
+export type TwitterUserResponse = {
+  data: TwitterUser
+}
+
+export type TwitterTimelineResponse = {
+  data: Tweet[];
+  meta: {
+      oldest_id: string;
+      newest_id: string;
+      result_count: number;
+      nextToken: string;
+  };
+  includes?: {
+    media?: TwitterMedia[]
+  }
+}
+
+export class TwitterService {
+  private bearerToken = Environment.TwitterBearerToken!;
+  private http = axios.create();
+
+  public constructor() {
```

Comment:
The way that we create services, is they are instantiated in `container.ts` then we called them from within the plugin with
```
this.container.twitterService...
```
Look at some other ones for an example